Code/GraphMol/Wrap/StereoGroup.cpp

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright (C) 2018 Dan Nealschneider
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#define NO_IMPORT_ARRAY
+#include <boost/python.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#include <string>
+
+// ours
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/StereoGroup.h>
+
+namespace python = boost::python;
+
+namespace RDKit {
+
+std::string stereoGroupClassDoc =
+    "A collection of atoms with a defined stereochemical relationship.\n\n"
+    "Used to help represent a sample with unknown stereochemistry, or that "
+    "is a mix\nof diastereomers.\n";
+
+struct stereogroup_wrap {
+  static void wrap() {
+    python::enum_<RDKit::StereoGroupType>(
+        "StereoGroupType")
+        .value("STEREO_ABSOLUTE", RDKit::StereoGroupType::STEREO_ABSOLUTE)
+        .value("STEREO_OR", RDKit::StereoGroupType::STEREO_OR)
+        .value("STEREO_AND", RDKit::StereoGroupType::STEREO_AND)
+        .export_values();
+
+    python::class_<ROMol::ATOM_PTR_VECT>("AtomVector")
+        .def(python::vector_indexing_suite<ROMol::ATOM_PTR_VECT>());
+
+    python::class_<StereoGroup, boost::shared_ptr<StereoGroup>> (
+        "StereoGroup1", stereoGroupClassDoc.c_str(), python::init<>())
+        .def("GetGroupType", &StereoGroup::getGroupType,
+             "Returns the StereoGroupType.\n")
+        .def("GetAtoms", &StereoGroup::getAtoms,
+             "Returns the StereoGroupType.\n",
+             python::return_internal_reference<
```

Comment:
I think you got the lifetime management here correct, but it's always a bit hard to tell (and if you didn't, it may not be directly fixable). Can you please add a test that

1. creates a molecule, `mol`, with a StereoGroup
2. gets a copy of that StereoGroup, `sg`
3. gets a copy of the StereoGroup's atoms, `sga`
4. reassigns `mol` to None
5. reassigns `sg` to None
6. tries to do something with the atoms in `sga`
7. tries to do something with owning mol of one of the atoms in `sga`.

The last two steps should, in theory, not cause problems. But who knows...

Reply:
huh. In a test, I did this:

    m = next(suppl)
    atoms = m.GetStereoGroups()[1].GetAtoms()
    m = None
    gc.collect()
    self.assertEqual(atoms[1].GetIdx(), 4)

This passed! I was expecting that the mol would be cleaned up (because I see nothing in the atom class that prevents that). I guess that this counts as "not causing problems", but I'm not sure now how to make sure that I'm not leaking mols. Maybe boost::python really is that cool, and takes care of it via that return_internal_reference business.

Maybe I'll use a hacked build with reference counting of ROMol to see what happens.

Reply:
Yeah, all that return_internal_reference and with_custodian_and_ward stuff really does work. It's magic. :-)

Reply:
I've mostly used SWIG - and this is _vastly_ more pleasant. The level of magic seems right, and like it is designed to go with these languages. pybind11 also seems like it might be nice, but maybe it's a bit early for that project.